creator-node/src/RehydrateIpfsQueue.js

```diff
@@ -3,14 +3,62 @@ const { rehydrateIpfsFromFsIfNecessary, rehydrateIpfsDirFromFsIfNecessary } = re
 const { logger: genericLogger } = require('./logging')
 const config = require('./config')
 const enableRehydrate = config.get('enableRehydrate')
+const redisClient = require('./redis')
 
 const PROCESS_NAMES = Object.freeze({
   rehydrate_dir: 'rehydrate_dir',
   rehydrate_file: 'rehydrate_file'
 })
 
+// minimum threshold per day to rehydrate
+const MIN_REHYDRATE_THRESHOLD = config.get('minRehydrateThreshold') || 10
 const MAX_COUNT = 10000
 
+class RehydrateRedisCounter {
+  constructor (redis) {
+    this.redis = redis
+  }
+
+  /**
+   * A redis key that stores hash of CID -> count and is incremented when we get a request for the CID
+   * A new key is created for each day and is automatically deleted 24 hours after the final write
+   * @param {String} taskType type of rehydrate task eg 'addRehydrateIpfsFromFsIfNecessaryTask' or 'addRehydrateIpfsDirFromFsIfNecessaryTask'
+   * @returns {String} redis key
+   */
+  static constructCounterRedisKey (taskType) {
+    return `${taskType}:::counter:::${new Date().toISOString().split('T')[0]}`
+  }
+
+  // keep count in redis for the number of times rehydrate is requested for each CID
+  async incrementCount (taskType, CID) {
+    const count = await this.redis.hincrby(RehydrateRedisCounter.constructCounterRedisKey(taskType), CID, 1)
```

Comment:
let's move `RehydrateRedisCounter.constructCounterRedisKey(taskType)` into the fn local scope as a const so we don't have to keep remake it again in line 35

Reply:
done